pkg/tsdb/tempo/tempo.go

```diff
@@ -73,20 +73,21 @@ func (e *tempoExecutor) Query(ctx context.Context, dsInfo *models.DataSource, ts
 
 	defer func() {
 		if err := resp.Body.Close(); err != nil {
-			plog.Warn("failed to close response body", "err", err)
+			tlog.Warn("failed to close response body", "err", err)
 		}
 	}()
 
-	if resp.StatusCode == http.StatusNotFound {
-		queryResult.Error = fmt.Errorf("failed to get trace: %s", traceID)
-		return result, nil
-	}
-
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
 
+	if resp.StatusCode != http.StatusOK {
+		queryResult.Error = fmt.Errorf("failed to get trace: %s", traceID)
```

Comment:
Sorry just now noticing this. I think we should send back also the underlying error so user knows if it's 500 or 404 of something else.